File: android/app/src/main/java/com/boxboxnow/app/models/DriverCard.kt
package com.boxboxnow.app.models

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowDownward
import androidx.compose.material.icons.filled.ArrowUpward
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.AvTimer
import androidx.compose.material.icons.filled.ViewAgenda
import androidx.compose.material.icons.filled.CompareArrows
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.FormatListNumbered
import androidx.compose.material.icons.filled.GpsFixed
import androidx.compose.material.icons.filled.HourglassBottom
import androidx.compose.material.icons.filled.KeyboardDoubleArrowDown
import androidx.compose.material.icons.filled.KeyboardDoubleArrowUp
import androidx.compose.material.icons.filled.Looks3
import androidx.compose.material.icons.filled.LooksOne
import androidx.compose.material.icons.filled.LooksTwo
import androidx.compose.material.icons.filled.MeetingRoom
import androidx.compose.material.icons.filled.MilitaryTech
import androidx.compose.material.icons.filled.MultilineChart
import androidx.compose.material.icons.filled.Numbers
import androidx.compose.material.icons.filled.OpenWith
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.ShowChart
import androidx.compose.material.icons.filled.Speed
import androidx.compose.material.icons.filled.Star
import androidx.compose.material.icons.filled.StarOutline
import androidx.compose.material.icons.filled.Timelapse
import androidx.compose.material.icons.filled.Timer
import androidx.compose.material.icons.filled.TrendingFlat
import androidx.compose.material.icons.outlined.EmojiEvents
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector

enum class DriverCardGroup(val label: String) {
    // Race split into Apex (raw live-timing values) and BBN (BoxBoxNow
    // analytics) so the pilot can tell at a glance whether a card
    // comes straight from Apex or from our own computations.
    RACE_APEX("Carrera - Apex"),
    RACE_BBN("Carrera - BBN"),
    BOX("BOX"),
    GPS("GPS"),
}

/**
 * The driver cards available in the pilot view. Order matches the BBN
 * indicator spreadsheet and is shared with the backend
 * (backend/app/services/driver_cards.py), web
 * (frontend/src/hooks/useDriverConfig.ts) and iOS (DriverCard.swift).
 */
enum class DriverCard(val key: String, val display: String, val sampleValue: String) {
    // --- Carrera - Apex (Excel order) ---
    RaceTimer("raceTimer", "Tiempo de carrera", "1:23:45"),
    StintTime("stintTime", "Tiempo de stint", "12:45"),
    BestStintLap("bestStintLap", "Mejor vuelta stint", "1:01.234"),
    LastLap("lastLap", "Ultima vuelta", "1:02.456"),
    ApexPosition("apexPosition", "Posición Apex", "P4/12"),
    TotalLaps("totalLaps", "Número de vueltas totales", "47"),
    StintLaps("stintLaps", "Número de vueltas stint", "12"),
    IntervalAhead("intervalAhead", "Intervalo kart delante", "0.968s"),
    IntervalBehind("intervalBehind", "Intervalo kart detrás", "0.973s"),
    /** Composite card: best S1 / S2 / S3 of this driver, 3 lines. */
    Sectors("sectors", "Mejores sectores", "S1 21.345"),
    BestS1("bestS1", "Mejor S1", "21.345"),
    BestS2("bestS2", "Mejor S2", "19.812"),
    BestS3("bestS3", "Mejor S3", "22.114"),

    // --- Carrera - BBN (Excel order) ---
    Position("position", "Posición (tiempos medios)", "P3/12"),
    AvgLap20("avgLap20", "Vuelta media (20v)", "1:03.120"),
    Best3("best3", "Media Mejor 3 v", "1:01.890"),
    AvgFutureStint("avgFutureStint", "Media stint futuro", "0:38:20"),
    TimeToMaxStint("timeToMaxStint", "Tiempo hasta stint máximo", "07:13"),
    LapsToMaxStint("lapsToMaxStint", "Vueltas hasta stint máximo", "5.2"),
    KartTier("kartTier", "Calificación del kart", "TIER 87"),
    TheoreticalBestLap("theoreticalBestLap", "Mejor vuelta teórica sectores", "1:01.67"),
    /** Combined view of S1/S2/S3 deltas in three lines on a single card. */
    DeltaSectors("deltaSectors", "Δ Sectores", "S1 -0.04s"),
    DeltaBestS1("deltaBestS1", "Δ Mejor S1", "+0.21s"),
    DeltaBestS2("deltaBestS2", "Δ Mejor S2", "-0.15s"),
    DeltaBestS3("deltaBestS3", "Δ Mejor S3", "+0.08s"),
    DeltaSectorsCurrent("deltaSectorsCurrent", "Δ Sectores Actual", "S1 +0.12s"),
    DeltaCurrentS1("deltaCurrentS1", "Δ Actual S1", "+0.18s"),
    DeltaCurrentS2("deltaCurrentS2", "Δ Actual S2", "-0.09s"),
    DeltaCurrentS3("deltaCurrentS3", "Δ Actual S3", "+0.31s"),
    RealPos("realPos", "Posición (clasif. real)", "P5/12"),
    GapAhead("gapAhead", "Gap Real Kart delante", "-1.2s"),
    GapBehind("gapBehind", "Gap Real Kart detrás", "+0.8s"),

    // --- Box (Excel order) ---
    CurrentPit("currentPit", "Pit en curso", "0:45"),
    BoxScore("boxScore", "Puntuación Box", "87"),
    PitCount("pitCount", "PITS (realizados / mínimos)", "2/4"),
    PitWindow("pitWindow", "Ventana de pit (open/closed)", "OPEN"),

    // --- GPS (Excel order). CurrentLapTime lives here because it needs
    //     a live GPS fix to be useful. ---
    DeltaBestLap("deltaBestLap", "Delta vs Best Lap (GPS)", "-0.32s"),
    GpsLapDelta("gpsLapDelta", "Delta vuelta anterior GPS", "+0.15s"),
    ProjectedLap("projectedLap", "Vuelta proyectada (GPS)", "1:01.45"),
    GForceRadar("gForceRadar", "G-Force (diana)", "G"),
    GpsGForce("gpsGForce", "G-Force (números)", "1.2G"),
    GpsSpeed("gpsSpeed", "Velocidad GPS", "94 km/h"),
    CurrentLapTime("currentLapTime", "Vuelta actual (tiempo real)", "0:42.318");

    val group: DriverCardGroup get() = when (this) {
        BoxScore, PitCount, CurrentPit, PitWindow -> DriverCardGroup.BOX
        CurrentLapTime, DeltaBestLap, GForceRadar, GpsLapDelta, ProjectedLap, GpsSpeed, GpsGForce -> DriverCardGroup.GPS
        RaceTimer, StintTime, BestStintLap, LastLap, ApexPosition, TotalLaps, StintLaps,
        IntervalAhead, IntervalBehind, Sectors, BestS1, BestS2, BestS3 -> DriverCardGroup.RACE_APEX
        else -> DriverCardGroup.RACE_BBN
    }

    val requiresGPS: Boolean get() = when (this) {
        CurrentLapTime, DeltaBestLap, GForceRadar, GpsLapDelta, ProjectedLap, GpsSpeed, GpsGForce -> true
        else -> false
    }

    val accent: Color get() = when (this) {
        RaceTimer -> Color(0xFF8E8E93)        // systemGray
        CurrentLapTime -> Color(0xFF2196F3)   // blue
        LastLap -> Color(0xFF8E8E93)          // systemGray
        DeltaBestLap -> Color(0xFF9C27B0)     // purple
        GForceRadar -> Color(0xFF8E8E93)      // systemGray
        Position -> Color(0xFF9C27B0)         // purple
        RealPos -> Color(0xFF41D238)          // accent
        GapAhead -> Color(0xFFFF453A)         // errorRed
        GapBehind -> Color(0xFF30D158)        // successGreen
        AvgLap20 -> Color(0xFF3F51B5)         // indigo
        Best3 -> Color(0xFFFF9F0A)            // warningOrange
        AvgFutureStint -> Color(0xFF00BFA5)   // teal
        BoxScore -> Color(0xFFFFCC00)         // yellow
        BestStintLap -> Color(0xFF9C27B0)     // purple
        GpsLapDelta -> Color(0xFF00BCD4)      // cyan
        ProjectedLap -> Color(0xFF00BCD4)     // cyan, same as GpsLapDelta
        GpsSpeed -> Color(0xFF2196F3)         // blue
        GpsGForce -> Color(0xFF34C759)        // emerald
        LapsToMaxStint -> Color(0xFF00BFA5)   // teal
        PitWindow -> Color(0xFF30D158)        // successGreen
        PitCount -> Color(0xFFFF9F0A)         // warningOrange
        CurrentPit -> Color(0xFF00BCD4)       // cyan
        DeltaBestS1, DeltaBestS2, DeltaBestS3 -> Color(0xFFFFCC00)  // yellow
        DeltaSectors -> Color(0xFFFFCC00)     // yellow
        DeltaCurrentS1, DeltaCurrentS2, DeltaCurrentS3 -> Color(0xFFFFCC00)
        DeltaSectorsCurrent -> Color(0xFFFFCC00)
        TheoreticalBestLap -> Color(0xFFFF4081) // pink
        IntervalAhead -> Color(0xFFFF453A)    // red, mirrors GapAhead
        IntervalBehind -> Color(0xFF30D158)   // green, mirrors GapBehind
        ApexPosition -> Color(0xFF9C27B0)     // purple
        // 2026-05 additions
        StintTime -> Color(0xFF8E8E93)        // gray
        TotalLaps -> Color(0xFF8E8E93)
        StintLaps -> Color(0xFF8E8E93)
        Sectors -> Color(0xFF9C27B0)          // purple
        BestS1, BestS2, BestS3 -> Color(0xFF9C27B0)
        TimeToMaxStint -> Color(0xFFFF9F0A)   // warningOrange
        KartTier -> Color(0xFF41D238)         // accent
    }

    val iconMaterial: ImageVector get() = when (this) {
        RaceTimer -> Icons.Filled.Timer
        CurrentLapTime -> Icons.Filled.AvTimer
        LastLap -> Icons.Filled.Timelapse
        DeltaBestLap -> Icons.Filled.CompareArrows
        GForceRadar -> Icons.Filled.GpsFixed
        Position -> Icons.Filled.EmojiEvents
        RealPos -> Icons.Outlined.EmojiEvents
        GapAhead -> Icons.Filled.ArrowUpward
        GapBehind -> Icons.Filled.ArrowDownward
        AvgLap20 -> Icons.Filled.MultilineChart
        Best3 -> Icons.Filled.Star
        AvgFutureStint -> Icons.Filled.ShowChart
        BoxScore -> Icons.Filled.Speed
        BestStintLap -> Icons.Filled.StarOutline
        GpsLapDelta -> Icons.Filled.TrendingFlat
        ProjectedLap -> Icons.Filled.TrendingFlat
        GpsSpeed -> Icons.Filled.Speed
        GpsGForce -> Icons.Filled.OpenWith
        LapsToMaxStint -> Icons.Filled.Refresh
        PitWindow -> Icons.Filled.MeetingRoom
        PitCount -> Icons.Filled.LooksOne
        CurrentPit -> Icons.Filled.AvTimer
        DeltaBestS1 -> Icons.Filled.LooksOne
        DeltaBestS2 -> Icons.Filled.LooksTwo
        DeltaBestS3 -> Icons.Filled.Looks3
        TheoreticalBestLap -> Icons.Filled.AutoAwesome
        IntervalAhead -> Icons.Filled.KeyboardDoubleArrowUp
        IntervalBehind -> Icons.Filled.KeyboardDoubleArrowDown
        ApexPosition -> Icons.Filled.FormatListNumbered
        DeltaSectors -> Icons.Filled.ViewAgenda
        DeltaCurrentS1 -> Icons.Filled.LooksOne
        DeltaCurrentS2 -> Icons.Filled.LooksTwo
        DeltaCurrentS3 -> Icons.Filled.Looks3
        DeltaSectorsCurrent -> Icons.Filled.ViewAgenda
        // 2026-05 additions
        StintTime -> Icons.Filled.AvTimer
        TotalLaps -> Icons.Filled.Numbers
        StintLaps -> Icons.Filled.Refresh
        Sectors -> Icons.Filled.ViewAgenda
        BestS1 -> Icons.Filled.LooksOne
        BestS2 -> Icons.Filled.LooksTwo
        BestS3 -> Icons.Filled.Looks3
        TimeToMaxStint -> Icons.Filled.HourglassBottom
        KartTier -> Icons.Filled.MilitaryTech
    }

    /** i18n catalog key for the card's user-visible label. Matches the
     *  web (`card.<id>`) and iOS so the three platforms share one set of
     *  translations. Render with `t(card.labelKey)` inside a Composable;
     *  `display` stays as the Spanish fallback for non-Composable code. */
    val labelKey: String get() = "card.$key"

    companion object {
        fun fromKey(key: String): DriverCard? = entries.firstOrNull { it.key == key }

        /** Cards visible by default; GPS-only cards off. */
        val defaultVisible: Map<String, Boolean> =
            entries.associate { it.key to !it.requiresGPS }

        /** Cards in catalog (Excel) order, grouped by category. Used for
         *  rendering the config checkboxes and as the default card order
         *  for new presets. Order matches the BBN indicator spreadsheet
         *  — NOT alphabetical by localized label, so the picker stays
         *  consistent across languages and aligned with the landing's
         *  comparison table. */
        val byGroupAndCatalogOrder: List<DriverCard> = buildList {
            for (group in DriverCardGroup.entries) {
                addAll(entries.filter { it.group == group })
            }
        }

        val defaultOrder: List<String> = byGroupAndCatalogOrder.map { it.key }
    }
}

File: BoxBoxNow/BoxBoxNow/Models/DriverCard.swift
import Foundation
import SwiftUI

enum DriverCardGroup: String, CaseIterable {
    // Race split into Apex (raw live-timing values) and BBN (BoxBoxNow
    // analytics) so the pilot can tell at a glance whether a card
    // comes straight from Apex or from our own computations. Card
    // `rawValue`s are unchanged — only the group classification and
    // a few display labels were tweaked to match the spec.
    case raceApex
    case raceBbn
    case box
    case gps

    var label: String {
        switch self {
        case .raceApex: return "Carrera - Apex"
        case .raceBbn:  return "Carrera - BBN"
        case .box:      return "BOX"
        case .gps:      return "GPS"
        }
    }
}

enum DriverCard: String, CaseIterable, Codable, Identifiable {
    // Carrera - Apex (raw Apex live timing). Order matches the BBN
    // indicator spreadsheet (App móvil · Tarjetas Carrera · Timing).
    case raceTimer
    case stintTime
    case bestStintLap
    case lastLap
    case apexPosition    // Raw Apex live timing position (e.g. "4/7")
    case totalLaps
    case stintLaps
    case intervalAhead   // Apex interval to kart in front (myKart.interval)
    case intervalBehind  // Apex interval reported by the kart behind me
    /// Composite card: best S1 / S2 / S3 of this driver combined into
    /// one card, 3 lines. Distinct from `deltaSectors` (vs field-best).
    case sectors
    case bestS1
    case bestS2
    case bestS3

    // Carrera - BBN (BoxBoxNow-derived analytics). Order matches the
    // BBN indicator spreadsheet (App móvil · Tarjetas Carrera · BBN).
    case position
    case avgLap20
    case best3
    case avgFutureStint
    case timeToMaxStint
    case lapsToMaxStint
    case kartTier
    case theoreticalBestLap
    // Sector delta cards. Auto-hide via `requiresSectors` when the
    // active session doesn't expose sectors.
    /// Combined view of S1/S2/S3 deltas in three lines on a single
    /// card — same colored-by-sign math as `deltaBestS1/2/3` but
    /// without the leader's kart number / team / driver. Saves grid
    /// real estate when the pilot wants the three-sector summary at
    /// a glance.
    case deltaSectors
    case deltaBestS1
    case deltaBestS2
    case deltaBestS3
    /// Combined view of S1/S2/S3 current-pass deltas in three lines,
    /// mirroring `deltaSectors` but reading `sectorMetaCurrent`.
    case deltaSectorsCurrent
    // Current-pass sector cards — compare the pilot's latest sector
    // against the fastest *live pass* among on-track karts
    // (sectorMetaCurrent). Purely additive; existing "Δ Mejor" cards
    // are unchanged.
    case deltaCurrentS1
    case deltaCurrentS2
    case deltaCurrentS3
    case realPos
    case gapAhead
    case gapBehind

    // Box (Excel order)
    case currentPit
    case boxScore
    case pitCount
    case pitWindow

    // GPS (Excel order). `currentLapTime` lives here because it needs a
    // GPS fix to be useful (without GPS we don't know "where in the lap"
    // the kart is).
    case deltaBestLap
    case gpsLapDelta
    case projectedLap
    case gForceRadar
    case gpsGForce
    case gpsSpeed
    case currentLapTime

    var id: String { rawValue }

    var group: DriverCardGroup {
        switch self {
        case .boxScore, .pitCount, .currentPit, .pitWindow:
            return .box
        case .currentLapTime, .deltaBestLap, .projectedLap, .gForceRadar, .gpsLapDelta, .gpsSpeed, .gpsGForce:
            // GPS group. `currentLapTime` lives here because it needs a
            // GPS fix to be useful.
            return .gps
        // Carrera - Apex: raw Apex live-timing values, no client-side
        // recomputation, mirrors what the pilot would see on Apex's
        // own live timing screen.
        case .raceTimer, .stintTime, .bestStintLap, .lastLap, .apexPosition,
             .totalLaps, .stintLaps, .intervalAhead, .intervalBehind,
             .sectors, .bestS1, .bestS2, .bestS3:
            return .raceApex
        default:
            // Carrera - BBN: BoxBoxNow-derived analytics (avg pace,
            // adjusted classification, sector deltas, future-stint
            // estimates, etc.).
            return .raceBbn
        }
    }

    /// i18n catalog key for this card's label — `card.<rawValue>`,
    /// the exact key set shared with web (lib/i18n.ts) and Android
    /// (Translations.kt / DriverCard.labelKey).
    var labelKey: String { "card.\(rawValue)" }

    /// Localized label. Routes through `t(_:)` so every existing call
    /// site (`card.displayName` in the grid header, config screens,
    /// accessibility) follows the active language with no further
    /// changes. `t()` falls back to the Spanish catalog entry, then to
    /// the key itself, so a missing translation never crashes the view.
    var displayName: String { t(labelKey) }

    var iconName: String {
        switch self {
        case .raceTimer:      return "timer"
        case .currentLapTime: return "stopwatch"
        case .lastLap:        return "clock"
        case .deltaBestLap:   return "plusminus"
        case .projectedLap:   return "flag.checkered"
        case .gForceRadar:    return "gyroscope"
        case .position:       return "trophy.fill"
        case .realPos:        return "trophy"
        case .gapAhead:       return "arrow.up"
        case .gapBehind:      return "arrow.down"
        case .avgLap20:       return "chart.line.uptrend.xyaxis"
        case .best3:          return "star.fill"
        case .avgFutureStint: return "clock.arrow.circlepath"
        case .boxScore:       return "gauge.medium"
        case .bestStintLap:   return "star"
        case .gpsLapDelta:    return "arrow.left.arrow.right"
        case .gpsSpeed:       return "gauge.open.with.lines.needle.33percent.and.arrowtriangle"
        case .gpsGForce:      return "move.3d"
        case .lapsToMaxStint: return "repeat"
        case .pitWindow:      return "door.left.hand.open"
        case .pitCount:       return "number.circle"
        case .currentPit:     return "stopwatch.fill"
        case .deltaBestS1:    return "1.circle.fill"
        case .deltaBestS2:    return "2.circle.fill"
        case .deltaBestS3:    return "3.circle.fill"
        case .theoreticalBestLap: return "wand.and.stars"
        case .intervalAhead:  return "chevron.up.2"
        case .intervalBehind: return "chevron.down.2"
        case .apexPosition:   return "list.number"
        case .deltaSectors:   return "square.stack.3d.up.fill"
        case .deltaCurrentS1: return "1.circle.fill"
        case .deltaCurrentS2: return "2.circle.fill"
        case .deltaCurrentS3: return "3.circle.fill"
        case .deltaSectorsCurrent: return "square.stack.3d.up.fill"
        // 2026-05 additions
        case .stintTime:        return "stopwatch.fill"
        case .totalLaps:        return "number"
        case .stintLaps:        return "repeat.circle"
        case .sectors:          return "rectangle.3.group"
        case .bestS1:           return "1.square.fill"
        case .bestS2:           return "2.square.fill"
        case .bestS3:           return "3.square.fill"
        case .timeToMaxStint:   return "hourglass"
        case .kartTier:         return "rosette"
        }
    }

    /// Whether this card requires GPS (RaceBox or phone) data.
    /// `deltaBestLap` falls back to server-based delta (last - best) when
    /// GPS isn't available, so it's not strictly GPS-required anymore.
    var requiresGPS: Bool {
        switch self {
        case .currentLapTime, .projectedLap, .gForceRadar, .gpsLapDelta, .gpsSpeed, .gpsGForce:
            return true
        default:
            return false
        }
    }

    /// Whether this card requires the active session to expose sector
    /// telemetry (Apex grid must declare `s1|s2|s3` data-type columns).
    /// On circuits without sectors, the sector cards still appear in
    /// the config picker but show "--" in place of values.
    var requiresSectors: Bool {
        switch self {
        case .deltaBestS1, .deltaBestS2, .deltaBestS3, .theoreticalBestLap, .deltaSectors,
             .deltaCurrentS1, .deltaCurrentS2, .deltaCurrentS3, .deltaSectorsCurrent,
             // Best-sector cards (single + composite) — only meaningful on
             // circuits whose Apex grid declares `s1|s2|s3` columns.
             .sectors, .bestS1, .bestS2, .bestS3:
            return true
        default:
            return false
        }
    }

    /// Sample value shown in card order preview
    var sampleValue: String {
        switch self {
        case .raceTimer:      return "1:23:45"
        case .currentLapTime: return "0:42.318"
        case .lastLap:        return "1:02.456"
        case .deltaBestLap:   return "-0.32s"
        case .projectedLap:   return "1:01.45"
        case .gForceRadar:    return "G"
        case .position:       return "P3/12"
        case .realPos:        return "P5/12"
        case .gapAhead:       return "-1.2s"
        case .gapBehind:      return "+0.8s"
        case .avgLap20:       return "1:03.120"
        case .best3:          return "1:01.890"
        case .avgFutureStint: return "0:38:20"
        case .boxScore:       return "87"
        case .bestStintLap:   return "1:01.234"
        case .gpsLapDelta:    return "+0.15s"
        case .gpsSpeed:       return "94 km/h"
        case .gpsGForce:      return "1.2G"
        case .lapsToMaxStint: return "5.2"
        case .pitWindow:      return "OPEN"
        case .pitCount:       return "2/4"
        case .currentPit:     return "0:45"
        case .deltaBestS1:    return "+0.21s"
        case .deltaBestS2:    return "-0.15s"
        case .deltaBestS3:    return "+0.08s"
        case .theoreticalBestLap: return "1:01.67"
        case .intervalAhead:  return "0.968s"
        case .intervalBehind: return "0.973s"
        case .apexPosition:   return "P4/12"
        case .deltaSectors:   return "S1 -0.04s"
        case .deltaCurrentS1: return "+0.18s"
        case .deltaCurrentS2: return "-0.09s"
        case .deltaCurrentS3: return "+0.31s"
        case .deltaSectorsCurrent: return "S1 +0.12s"
        // 2026-05 additions
        case .stintTime:        return "12:45"
        case .totalLaps:        return "47"
        case .stintLaps:        return "12"
        case .sectors:          return "S1 21.345"
        case .bestS1:           return "21.345"
        case .bestS2:           return "19.812"
        case .bestS3:           return "22.114"
        case .timeToMaxStint:   return "07:13"
        case .kartTier:         return "TIER 87"
        }
    }

    /// Accent color for each card (matching web CARD_ACCENTS)
    var accentColor: Color {
        switch self {
        case .raceTimer:      return .gray
        case .currentLapTime: return .blue
        case .lastLap:        return .gray
        case .deltaBestLap:   return .purple
        case .projectedLap:   return .cyan
        case .gForceRadar:    return .gray
        case .position:       return .purple
        case .realPos:        return .accentColor
        case .gapAhead:       return .red
        case .gapBehind:      return .green
        case .avgLap20:       return .indigo
        case .best3:          return .orange
        case .avgFutureStint: return .teal
        case .boxScore:       return .yellow
        case .bestStintLap:   return .purple
        case .gpsLapDelta:    return .cyan
        case .gpsSpeed:       return .blue
        case .gpsGForce:      return .green
        case .lapsToMaxStint: return .teal
        case .pitWindow:      return .green
        case .pitCount:       return .orange
        case .currentPit:     return .cyan
        case .deltaBestS1, .deltaBestS2, .deltaBestS3: return .yellow
        case .theoreticalBestLap: return .pink
        case .intervalAhead:   return .red    // matches gapAhead semantics
        case .intervalBehind:  return .green  // matches gapBehind semantics
        case .apexPosition:    return .purple // distinct from "position" (.purple) and "realPos" (.accentColor)
        case .deltaSectors:    return .yellow // same family as deltaBestS1/2/3
        case .deltaCurrentS1, .deltaCurrentS2, .deltaCurrentS3: return .yellow
        case .deltaSectorsCurrent: return .yellow
        // 2026-05 additions
        case .stintTime:        return .gray
        case .totalLaps:        return .gray
        case .stintLaps:        return .gray
        case .sectors:          return .purple
        case .bestS1, .bestS2, .bestS3: return .purple
        case .timeToMaxStint:   return .orange
        case .kartTier:         return .accentColor
        }
    }

    static let defaultVisible: [String: Bool] = {
        var dict = [String: Bool]()
        for card in DriverCard.allCases {
            // GPS cards stay off by default (require RaceBox or phone GPS).
            // Sector cards default to ON: they self-handle "no sector data"
            // by showing "--" on circuits without S1/S2/S3 columns, so the
            // pilot doesn't have to know whether the active circuit
            // exposes them — the cards just light up where it matters.
            dict[card.rawValue] = !card.requiresGPS
        }
        return dict
    }()

    static let defaultOrder: [String] = DriverCard.allCases.map { $0.rawValue }
}
